src/@next/Switch/SwitchStyle.tsx

```diff
@@ -67,13 +74,25 @@ export const SwitchStyle = styled.div`
     right: 0;
     border-radius: 16px;
     background: ${Neutral.B100};
-    box-shadow: 0px 1px 3px rgba(0, 0, 0, 0.1);
+    box-shadow: 0 1px 3px rgba(0, 0, 0, 0.1);
     transform: scale(1);
     @media (max-width: ${Breakpoints.large}) {
       width: 14px;
       height: 14px;
     }
   }
+
+  &[data-loading='true']::before {
+    border: 2px solid ${Neutral.B100};
+    border-top-color: transparent;
+    background: transparent;
+    animation: ${spin} 0.6s linear infinite;
+    transform: rotate(0deg);
+  }
+
+  &[data-loading='true'] {
+    pointer-events: none;
+  }
```

Comment:
please add test cases for the loading state of the switch

Reply:
Done. I added one more commit with switch variants.